### PR TITLE
Nick

### PR DESCRIPTION
**Pull request by nwestallen**

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -1 +1 @@
...
...10,"nodeType":"38","messageId":"39","endLine":5,"endColumn":19},{"ruleId":"36","severity":1,"message":"43","line":8,"column":18,"nodeType":"38","messageId":"39","endLine":8,"endColumn":24},{"ruleId":"36","severity":1,"message":"44","line":55,"column":9,"nodeType":"38","messageId":"39","endLine":55,"endColumn":21},{"ruleId":"36","severity":1,"message":"45","line":2,"column":17,"nodeType":"38","messageId":"39","endLine":2,"endColumn":25},{"ruleId":"36","severity":1,"message":"42","line":2,"column":27,"nodeType":"38","messageId":"39","endLine":2,"endColumn":36},{"ruleId":"36","severity":1,"message":"37","line":3,"column":10,"nodeType":"38","messageId":"39","endLine":3,"endColumn":14},{"ruleId":"36","severity":1,"message":"46","line":3,"column":16,"nodeType":"38","messageId":"39","endLine":3,"endColumn":24},"@typescript-eslint/no-unused-vars","'Card' is defined but never used.","Identifier","unusedVar","'CardBack' is defined but never used.","'Hand' is defined but never used.","'useEffect' is defined but never used.","'uuidv4' is defined but never used.","'defHandStyle' is assigned a value but never used.","'useState' is defined but never used.","'Cardback' is defined but never used."]
```
Comment (Jaya-Hoyt) [inline .eslintcache]: This file should be in the .gitignore file. We don’t want to track it.

File: src/constants.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+//Constants for initial useState, including typical 52 card deck
+const deck52 = [
```
Comment (Jaya-Hoyt) [inline src/constants.tsx:2]: This should be generated in a nested for loop.

we should also make an enumeration for the suits and another for the ranks so we can have more readable names when writing code or testing.

File: src/App.tsx [hunk 3]
```diff
@@ -7,8 +7,13 @@ import { useEffect, useState } from 'react';
 import { db } from './firebase';
 import { uuid as uuidv4 } from 'uuid';
 
+import { deck52 } from './constants';
+import Board from './components/Board';
+
 function App() {
   const [displayText, setDisplayText] = useState('no data yet');
+  const [room, setRoom] = useState({users: {0:{}}, pot: 0, board:[{}], deck: deck52})
```
Comment (Jaya-Hoyt) [inline src/App.tsx]: Type this with an interface

File: src/App.tsx [hunk 4]
```diff
@@ -7,63 +8,91 @@ import { useEffect, useState } from 'react';
...
-    button: 0,
-    blindSize: .02,
-    buyIn: 10,
-    actor: 0,
+useEffect( () => {
+  docRef.onSnapshot((doc) => {
+    if(doc.exists) {
+      console.log('Firebase data:', doc.data());
+      let data = doc.data();
+      if(data) {setRoom({
```
Comment (Jaya-Hoyt) [inline src/App.tsx:36]: You need a line break after the first curly brace

Comment (Jaya-Hoyt) [reply src/App.tsx:36]: We need eslint to be helping you out here.

File: src/App.tsx [hunk 5]
```diff
@@ -7,63 +8,91 @@ import { useEffect, useState } from 'react';
...
+  const dealCard = (position) => {
+    if (position === 'board') {
+      return docRef.update({
+        board: firebase.firestore.FieldValue.arrayUnion(room.deck[0]),
+        deck: firebase.firestore.FieldValue.arrayRemove(room.deck[0])
+      })
+      .then(() => console.log('board updated'))
+      .catch((err) => console.log(err));
+    } else {
+      let playerHand = `users.${position}.cards`
```
Comment (Jaya-Hoyt) [inline src/App.tsx:62]: Use const whenever possible

File: src/App.tsx [hunk 6]
```diff
@@ -7,63 +8,91 @@ import { useEffect, useState } from 'react';
...
-          { rank: 11, suit: 2 },
-          ]} hidden={false} style={defHandStyle}
-        />
-        <p>
-        {displayText}
-        </p>
-      </header>
+      <Board boardCards={room.board} clearCards = {clearCards} dealCard={() => dealCard('board')} pot={room.pot} />
+      <PlayArea>
+        {Object.keys(room.users).map(key => <Seat hand={room.users[key].cards} dealCard={() => dealCard(key)} clearHand={() => clearHand(key)} player={room.users[key]} />)}
```
Comment (Jaya-Hoyt) [inline src/App.tsx:93]: Too much for one line. Make a line break after the opening paren for .map

use `Object.entries` to key both the key and value
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

File: src/components/Board.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+//Board component
+import React, { useState, useEffect } from 'react';
+import { Card, Cardback, Hand } from 'react-deck-o-cards';
+import { defHandStyle } from '../constants';
+
+const Board = (props) => {
```
Comment (Jaya-Hoyt) [inline src/components/Board.tsx:6]: Make an interface for your props in all your functional components that take props in

**Review by Jaya-Hoyt — CHANGES_REQUESTED** [review]
We need a good eslint config that can enforce a good style. If we already have one, then in `npx eslint src` to see where the warnings and errors are. Try to set up your editor to integrate with our eslint config so it helps you as you code.

**Comment by nwestallen** [comment]
I'm going to start a fresh branch so I can implement things via cloud functions and have a more readable working tree

**Pull request closed without merging**